### PR TITLE
Add a way to filter opt-pipeline passes

### LINKED ISSUE #6569: [REQUEST]: [LLVM]Add ability to search pass name in Opt pass pipeline view
### Is your feature request related to a problem? Please describe

Opt pass pipeline is a great feature but the number of passes are quite large and it is cumbersome to scroll. 

### Describe the solution you'd like

It would be nice if you can add a search box in the panel to allow users to search the pass name and jump to it.

### Describe alternatives you've considered

NA

### Additional context

Not applicable

### PR DESCRIPTION
**Pull request by RubenRBS**
Somewhat of a rough sketch in the matching department, I'd like something more like TomSelect does with the fuzzy matching, but need to look into getting that behaviour without using the select iself

Closes #6569 

(Else, we could always just move this to be a multi selector no?)

### CODE REVIEW

**Comment by RubenRBS** [comment]
image
Could look better, but it also could look worse!
image

**Review by jeremy-rifkin — APPROVED** [review]
I think this looks OK to me, haven't tested locally. Should the filter be part of the pane state?

> we could always just move this to be a multi selector

Do you refer to making the whole pass list a multi-selector? I wouldn't be a fan of that personally, I have found that a bit cumbersome on the tree/rtl viewer and one nice thing about the implementation here is you can use arrow keys to quickly browse the passes.

As far as fuzzy search logic, I more or less implemented tom-select's search logic for the compiler picker popup and it might be easy to extract that out and reuse it here.

**Comment by RubenRBS** [comment]
Thanks for the insight @jeremy-rifkin, I've implemented the sifter logic, works like a charm :)

**Review by jeremy-rifkin — APPROVED** [review]
I think LGTM

File: static/panes/opt-pipeline.ts [hunk 1]
```diff
@@ -131,6 +135,7 @@ export class OptPipeline extends MonacoPane<monaco.editor.IStandaloneDiffEditor,
         this.eventHub.emit('optPipelineViewOpened', this.compilerInfo.compilerId);
         this.eventHub.emit('requestSettings');
         this.emitOptions(true);
+        this.passesFilter.on('input', _.debounce(this.onFiltersChange.bind(this), 500));
```
Comment (RubenRBS) [inline static/panes/opt-pipeline.ts:138]:
```suggestion
        this.passesFilter.on('input', _.debounce(this.onFiltersChange.bind(this), 250));
```

**Pull request merged**